### PR TITLE
chore: Use SQLAlchemy 2 and support `psycopg[binary]`

### PR DESCRIPTION
**Pull request by edgarrmondragon**

### CODE REVIEW

**Comment by WillDaSilva** [comment]
This will supersede (unless we feel like updating it in `main` before Meltano v3 is released):
- #7546

**Comment by edgarrmondragon** [comment]
Having a few problems testing mssql locally: https://github.com/pymssql/pymssql/issues/822

File: pyproject.toml [hunk 1]
```diff
@@ -61,6 +61,7 @@ jsonschema = "^4.17"
 packaging = "^23.0"
 platformdirs = "^3.0.0"
 psutil = "^5.9.5"
+psycopg = {version = "^3.1.8", extras=["binary"]}
 psycopg2-binary = "^2.8.5"
```
[the remark this answers is not in the document]
Comment (WillDaSilva) [reply pyproject.toml:65]: We'll have to note this in the V2 -> V3 migration guide:
- https://github.com/meltano/meltano/issues/7957

**Comment by edgarrmondragon** [comment]
I'm splitting this PR into 3 smaller changes to make it easier to debug what's causing MSSQL to fail:

- https://github.com/meltano/meltano/pull/7960
- Bump SQLAlchemy to 2.0
- Add new `postgres` extra and `psycopg[binary]` dependency

**Comment by edgarrmondragon** [comment]
All items above have been addressed in their own PRs

**Pull request closed without merging**